File: Week 6 - Python/Problem Set 6/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    L = countLetters(text) / countWords(text) * 100
    S = countSentences(text) / countWords(text) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")

    elif index > 16:
        print("Grade 16+")

    else:
        print("Grade", index)


def countLetters(text):
    totalLetters = 0
    for char in text:
        if char.isalnum():
            totalLetters += 1

    return totalLetters


def countWords(text):
    return text.count(" ") + 1


def countSentences(text):
    totalSentences = 0
    for char in text:
        if char in ["!", ".", "?"]:
            totalSentences += 1

    return totalSentences


main()

File: Week 2 - Arrays/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float L = (float) countLetters(text) / (float) countWords(text) * 100;
    float S = (float) countSentences(text) / (float) countWords(text) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %i\n", index);
}

int countLetters(string text)
{
    int totalLetters = 0;
    for (int currentChar = 0, textLength = strlen(text); currentChar < textLength; currentChar++)
    {
        if (isalpha(text[currentChar]))
        {
            totalLetters++;
        }
    }
    return totalLetters;
}

int countWords(string text)
{
    int totalWords = 1;
    for (int currentChar = 0, textLength = strlen(text); currentChar < textLength; currentChar++)
    {
        if (isspace(text[currentChar]))
        {
            totalWords++;
        }
    }
    return totalWords;
}

int countSentences(string text)
{
    int totalSentences = 0;
    for (int currentChar = 0, textLength = strlen(text); currentChar < textLength; currentChar++)
    {
        if (text[currentChar] == 33 || text[currentChar] == 46 || text[currentChar] == 63)
        {
            totalSentences++;
        }
    }
    return totalSentences;
}
